File: apps/web/src/lib/webvtt.test.ts
import { describe, expect, it } from 'vitest';
import { parseWebVtt } from './webvtt';

describe('parseWebVtt', () => {
  it('parses cues with optional identifiers', () => {
    const cues = parseWebVtt(`WEBVTT

1
00:00:01.000 --> 00:00:04.000
Hello world

00:01:00.500 --> 00:01:02.000
Second line
`);
    expect(cues).toEqual([
      { startSeconds: 1, endSeconds: 4, text: 'Hello world' },
      { startSeconds: 60.5, endSeconds: 62, text: 'Second line' },
    ]);
  });

  it('strips simple tags and ignores NOTE blocks', () => {
    const cues = parseWebVtt(`WEBVTT

NOTE skip me

00:00:00.000 --> 00:00:01.000
<c>Hi</c> there
`);
    expect(cues).toEqual([{ startSeconds: 0, endSeconds: 1, text: 'Hi there' }]);
  });
});

File: apps/mobile/test/unit/webvtt_test.dart
import 'package:flutter_test/flutter_test.dart';
import 'package:forge_mobile/core/utils/webvtt.dart';

void main() {
  test('parseWebVtt extracts cues and strips markup', () {
    const source = '''
WEBVTT

00:00:01.000 --> 00:00:03.000
Hello <b>world</b>

00:00:04.000 --> 00:00:06.000
Second cue
''';
    final cues = parseWebVtt(source);
    expect(cues, hasLength(2));
    expect(cues[0].startSeconds, 1);
    expect(cues[0].text, 'Hello world');
    expect(cues[1].text, 'Second cue');
  });

  test('formatCueTimestamp pads minutes', () {
    expect(formatCueTimestamp(65), '1:05');
    expect(formatCueTimestamp(3661), '1:01:01');
  });

  test('activeCueTextAt returns the cue covering the playhead', () {
    final cues = parseWebVtt('''
WEBVTT

00:00:01.000 --> 00:00:03.000
Hello

00:00:04.000 --> 00:00:06.000
Second
''');
    expect(activeCueTextAt(cues, 0.5), isNull);
    expect(activeCueTextAt(cues, 1.5), 'Hello');
    expect(activeCueTextAt(cues, 3.0), isNull);
    expect(activeCueTextAt(cues, 4.5), 'Second');
  });
}
